src/v2/Utils/Hooks/__tests__/useFadeTransition.jest.tsx

```diff
@@ -0,0 +1,63 @@
+import { render } from "@testing-library/react"
+import { renderHook } from "@testing-library/react-hooks"
+import { useFadeTransition } from "../useFadeTransition"
+
+jest.mock("../../wait", () => ({
+  wait: () => Promise.resolve(),
+}))
+
+describe("useFadeTransition", () => {
+  it("sets some defaults", () => {
+    const { result } = renderHook(() => useFadeTransition())
+
+    expect(result.current.mode).toEqual("Resting")
+    expect(result.current.status).toEqual("In")
+  })
+
+  it("registers elements", () => {
+    const { result } = renderHook(() => useFadeTransition())
+
+    const Example = () => {
+      return (
+        <div>
+          <div ref={result.current.register(0)}>one</div>
+          <div ref={result.current.register(2)}>three</div>
+          <div ref={result.current.register(1)}>two</div>
+        </div>
+      )
+    }
+
+    render(<Example />)
+
+    expect(
+      result.current.refs.current.map(ref => ref.current!.innerHTML)
+    ).toEqual(["one", "two", "three"])
+  })
+
+  it("transitions elements", async () => {
+    const { result } = renderHook(() => useFadeTransition({ duration: 250 }))
+
+    const Example = () => {
+      return (
+        <div>
+          <div ref={result.current.register(0)}>one</div>
+          <div ref={result.current.register(2)}>three</div>
+          <div ref={result.current.register(1)}>two</div>
+        </div>
+      )
+    }
+
+    render(<Example />)
+
+    await result.current.transition("Out")
```

Comment:
This changes the order, right? I was expecting to see an assertion like this:

```
    expect(
      result.current.refs.current.map(ref => ref.current!.innerHTML)
    ).toEqual(["two", "three", "one"])
```

Or maybe I'm not understanding?

Reply:
I guess I could have written the spec differently. You'd expect them to be in order because I'm calling them with the correct indexes. I'm executing the calls out of order: `register(0)`, `register(2)`, `register(1)`.